R/table.R

```diff
@@ -5,18 +5,24 @@
 
 # writing a dataFrame to a csv
 saveToCsv <- function(dataFrame, filePath) {
-  for (i in 1:dim(dataFrame)[2]) {
-    if (is.numeric(dataFrame[[i]])) {
-      dataFrame[[i]][is.nan(dataFrame[[i]])]<-"NaN"
-    } else if (is(dataFrame[[i]], "POSIXct")) {
-      # convert POSIXct before uploading to Synapse
-      dataFrame[[i]]<-format(as.POSIXlt(dataFrame[[i]], 'UTC', usetz=TRUE), "%Y-%m-%d %H:%M:%S.000")
+  if (is.data.frame(dataFrame) && nrow(dataFrame) != 0) {
+    for (i in 1:dim(dataFrame)[2]) {
+      if (is.numeric(dataFrame[[i]])) {
+        dataFrame[[i]][is.nan(dataFrame[[i]])]<-"NaN"
+      } else if (is(dataFrame[[i]], "POSIXct")) {
+        # convert POSIXct before uploading to Synapse
+        dataFrame[[i]]<-format(as.POSIXlt(dataFrame[[i]], 'UTC', usetz=TRUE), "%Y-%m-%d %H:%M:%S.000")
+      }
     }
+    write.csv(x=dataFrame, file=filePath, row.names=FALSE, na="")
```

Comment:
I'm concerned that the behavior for an empty data frame is not as expected. The behavior is that saveToCsv does nothing while one would expect that it writes a file having no data rows.